docs/script.rst

```diff
@@ -215,6 +215,40 @@ In the preceding example, ``blastp`` and its ``-in``, ``-out``, ``-db`` and ``-h
 their arguments are effectively a single line.
 
 
+Functions
+---------
+
+Functions can be defined using the following syntax::
+
+    def <function name> ( arg1, arg, .. ) {
+        <function body>
+    }
+
+For example::
+
+    def foo() {
+        'Hello world'
+    }
+
+    def bar(alpha, omega) {
+        alpha + omega
+    }
+
+The above snippet defines two simple functions, that can be invoked in the workflow script as ``foo()``, which
+returns ``'Hello world'``, and ``bar(10,20)``, which returns the sum of two parameters (``30`` in this case).
+
+.. note:: Functions implicitly return the result of the last evaluated statement.
+
+The keyword ``return`` can be used to explicitly exit from a function and return the specified value. For example::
```

Comment:
Is this meant to be part of the note above? I think either it should all be part of the same note block, or the sentence above should be merged into this paragraph without a note.

Reply:
Good point, I think it could be a single paragraph.